### PR TITLE
Feature/ami

### PR DESCRIPTION
**Pull request by discobean**
This adds an `${ami xxx}` feature that resolves Issue #350 

It still needs to be:
1. documented
2. tests completed
3. reviewed

Apart from that the lookup is working for us in prod.

### CODE REVIEW

**Comment by phobologic** [comment]
Hey @discobean - it looks like the change to the regexp actually caused an issue with nested regular expressions (it causes it to hang forever in the test: 

```
test_variable_resolve_nested_lookup (stacker.tests.test_variables.TestVariables
```

Doing nested regexps is a pain, really need to find a better way to do this.

**Comment by discobean** [comment]
Hi mate, I've just added [] to the regex, which supports our use case for the AMI lookup

**Comment by phobologic** [comment]
@discobean I went ahead and added some tests & docs for this lookup.  Can you take a look and let me know if that looks right to you?  Thanks!

File: stacker/lookups/__init__.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
                                        # space
 ?\s*                                   # any number of spaces separating the
                                        # type from the input
-(?P<input>[@\+\/,\._\-a-zA-Z0-9\:\s=]+) # the input value to the lookup
+(?P<input>[@\+\/,\._\-a-zA-Z0-9\:\s=\[\]\*]+) # the input value to the lookup
```
Comment (ralph-tice) [inline stacker/lookups/__init__.py]: i think this deserves a little more explanation or to be broken up further

Comment (phobologic) [inline stacker/lookups/__init__.py]: Honestly, @ralph-tice, I'd like to do away with that regex someday.  I've attempted to refactor the way we parse lookups in configs a few times now, and it's surprisingly hard for a few reasons:

- we allow nested lookups
- we allow multiple lookups per variable
- we allow lookups to return non-text values (basically any python object)
- lookups can actually affect the way the external code works (only the `output` lookup at this point, which is used to determine dependencies)

This results in it being shockingly complex to use one of the existing parsers out there.  At some point I'll give it another shot and hopefully be more successful.

**Comment by phobologic** [comment]
ping @discobean

**Comment by discobean** [comment]
Thanks @phobologic been away and didn't realised you brought this in.  Much appreciated.

**Comment by phobologic** [comment]
NP @discobean - thank you for the PR :)

**Pull request merged**